cumulusci/tasks/bulkdata/mapping_parser.py

```diff
@@ -124,3 +293,36 @@ def validate_mapping(cls, values):
 def parse_from_yaml(source: Union[str, Path, IO]) -> Dict:
     "Parse from a path, url, path-like or file-like"
     return MappingSteps.parse_from_yaml(source)
+
+
+def validate_mapping(
+    *,
+    mapping: Dict,
+    org_config: OrgConfig,
+    namespace: str,
+    data_operation: DataOperationType,
+    inject_namespaces: bool,
+    drop_missing: bool,
+):
+    should_continue = [
+        m.validate_and_inject_namespace(
+            org_config, namespace, data_operation, inject_namespaces, drop_missing
+        )
+        for m in mapping.values()
+    ]
+
+    if not drop_missing and not all(should_continue):
+        raise BulkDataException("One or more permissions errors blocked the operation.")
+
+    if drop_missing:
+        # Drop any steps with sObjects that are not present.
+        for (include, step_name) in zip(should_continue, list(mapping.keys())):
+            if not include:
+                del mapping[step_name]
+
+        # Remove any remaining lookups to dropped objects.
```

Comment:
Should we try to detect and error if one of these lookups is required?

Reply:
If the child object is a master-detail child, the parent must be visible or the child won't be. If it's a lookup relationship, though, that's a risk. We can check for database-level required fields, and I'll add that.

Reply:
Added!